src/IntegrationTests/ExplicitExpansion/DisableExplicitExpansion.cs

```diff
@@ -0,0 +1,111 @@
+
+        public string GetLastSelectSqlLogEntry() => Log.LastOrDefault(_ => _.Contains("SELECT"));
+    }
+
+    private static readonly IQueryable<Source> _iq = new List<Source> {
+        new Source() { Name = "Name1", Code = "Code1", Desc = -12 },
+    } .AsQueryable();
+
+    private static readonly Source _iqf = _iq.First();
+
+    public class DatabaseInitializer : DropCreateDatabaseAlways<Context>
+    {
+        protected override void Seed(Context context)
+        {
+            context.Sources.Add(_iqf);
+            base.Seed(context);
+        }
+    }
+
+    protected override MapperConfiguration CreateConfiguration() => new(cfg =>
+    {
+        cfg.CreateMap<Source, Dto>()
+            .ForMember(dto => dto.Code, conf => conf.ExplicitExpansion(false))
+            .ForAllMembers(conf => conf.ExplicitExpansion())
+            ;});
+
+    [Fact]
+    public void Should_CodeBeExpanded() {
+        using (var ctx = new Context()) {
+            var dto = ProjectTo<Dto>(ctx.Sources).ToList().First();
+            var sqlSelect = ctx.GetLastSelectSqlLogEntry();
+            sqlSelect.SqlFromShouldStartWith(nameof(ctx.Sources));
+            sqlSelect.ShouldNotContain("JOIN");
+
+            sqlSelect.SqlShouldNotSelectColumn(nameof(_iqf.Name)); dto.Name.ShouldBeNull();
+            sqlSelect.SqlShouldSelectColumn(nameof(_iqf.Code)); dto.Code.ShouldBe(_iqf.Code);
+            sqlSelect.SqlShouldNotSelectColumn(nameof(_iqf.Desc)); dto.Desc.ShouldBeNull();
+        }
+    }
+
+    [Fact]
+    public void Should_NameAndCodeBeExpanded() {
+        using (var ctx = new Context()) {
+            var dto = ProjectTo<Dto>(ctx.Sources, null, _ => _.Name).First();
+            var sqlSelect = ctx.GetLastSelectSqlLogEntry();
+            sqlSelect.SqlFromShouldStartWith(nameof(ctx.Sources));
+            sqlSelect.ShouldNotContain("JOIN");
+            
+            dto.Name.ShouldBe(_iqf.Name); sqlSelect.SqlShouldSelectColumn(nameof(_iqf.Name));
+            dto.Code.ShouldBe(_iqf.Code); sqlSelect.SqlShouldSelectColumn(nameof(_iqf.Code));
+            dto.Desc.ShouldBeNull(); sqlSelect.SqlShouldNotSelectColumn(nameof(_iqf.Desc));
+        }
+    }
+
+}
+
+public class ConstructorNoExplicitExpansion : IntegrationTest<ConstructorNoExplicitExpansion.DatabaseInitializer> {
+    public class Entity {
+        public int Id { get; set; }
+        public string Name { get; set; }
+    }
+    record Dto(string Name) { }
+    public class Context : LocalDbContext {
+        public DbSet<Entity> Entities { get; set; }
+    }
+    public class DatabaseInitializer : DropCreateDatabaseAlways<Context> {
+        protected override void Seed(Context context) {
+            context.Entities.Add(new() { Name = "Name" });
+            base.Seed(context);
+        }
+    }
+    protected override MapperConfiguration CreateConfiguration() => new(c => c.CreateProjection<Entity, Dto>().ForCtorParam("Name", o => o.ExplicitExpansion(false)));
```

Comment:
This should use mapping inheritance, the same as the test below.

Reply:
@lbargaoanu i added a new test for constructor mapping with inheritance (sorry for the long delay i needed to change my computer)